torch/_meta_registrations.py

```diff
@@ -291,19 +291,29 @@ def meta_fft_c2c(self, dim, normalization, forward):
     return output
 
 
+# fft_r2c meta fix by Juan Ceresa and Julian Mueller
 @register_meta([aten._fft_r2c.default, aten._fft_r2c.out])
 @out_wrapper()
 def meta_fft_r2c(self, dim, normalization, onesided):
     assert self.dtype.is_floating_point
     output_sizes = list(self.size())
 
+    last_dim = dim[-1]
+
     if onesided:
-        last_dim = dim[-1]
         last_dim_halfsize = (output_sizes[last_dim] // 2) + 1
         output_sizes[last_dim] = last_dim_halfsize
 
+    complex_dtype = self.dtype.to_complex()
+
+    input_strides = self.stride()
+    output_strides = list(input_strides)
+
+    if onesided:
+        output_strides[last_dim] = 1
+
     return self.new_empty(
-        output_sizes, dtype=utils.corresponding_complex_dtype(self.dtype)
+        output_sizes, dtype=complex_dtype, strides=output_strides
```

Comment:
what is the test plan for this PR?  I was expecting to see some modifications to existing unit tests - perhaps removal of 'skip' markers for tests that used to be broken and now passing.  Or, addition of new tests as needed.